tests/test_level/test_level_17.py

```diff
@@ -437,3 +448,147 @@ def test_not_equal_with_diff_types_gives_error(self, left, right):
       code=code,
       exception=exceptions.InvalidTypeCombinationException
     )
+
+  #
+  # if pressed turtle tests
+  #
+  def test_if_pressed_repeat_multiple_x_turtle_move(self):
+      code = textwrap.dedent("""\
+      repeat 10 times
+          if w is pressed:
+              forward 25
+          if a is pressed:
+              turn -90
+          if d is pressed:
+              turn 90
+          if s is pressed:
+              turn 180""")
+
+      expected = HedyTester.dedent("""\
+      for i in range(int('10')):
+        while not pygame_end:
+          pygame.display.update()
+          event = pygame.event.wait()
+          if event.type == pygame.QUIT:
+            pygame_end = True
+            pygame.quit()
+            break
+          if event.type == pygame.KEYDOWN: 
+            if event.key == pygame.K_w:
+              trtl = 25.0
+              try:
+                trtl = float(trtl)
+              except ValueError:
+                raise Exception(f'While running your program the command <span class="command-highlighted">forward</span> received the value <span class="command-highlighted">{trtl}</span> which is not allowed. Try changing the value to a number.')
+              t.forward(min(600, trtl) if trtl > 0 else max(-600, trtl))
+              time.sleep(0.1)
+              break
+            if event.key == pygame.K_a:
+              trtl = -90.0
+              try:
+                trtl = float(trtl)
+              except ValueError:
+                raise Exception(f'While running your program the command <span class="command-highlighted">turn</span> received the value <span class="command-highlighted">{trtl}</span> which is not allowed. Try changing the value to a number.')
+              t.right(min(600, trtl) if trtl > 0 else max(-600, trtl))
+              break
+            if event.key == pygame.K_d:
+              trtl = 90.0
+              try:
+                trtl = float(trtl)
+              except ValueError:
+                raise Exception(f'While running your program the command <span class="command-highlighted">turn</span> received the value <span class="command-highlighted">{trtl}</span> which is not allowed. Try changing the value to a number.')
+              t.right(min(600, trtl) if trtl > 0 else max(-600, trtl))
+              break
+            if event.key == pygame.K_s:
+              trtl = 180.0
+              try:
+                trtl = float(trtl)
+              except ValueError:
+                raise Exception(f'While running your program the command <span class="command-highlighted">turn</span> received the value <span class="command-highlighted">{trtl}</span> which is not allowed. Try changing the value to a number.')
```

Comment:
To transpile turtle commands you can use `Tester.py::turtle_color_command_transpiled`

Reply:
Thank you for point this out, commit [Make use of transpiled turtle commands for tests](https://github.com/hedyorg/hedy/pull/3483/commits/3274ab798a0e892a9289f3593e5146d629729ada) replaces all turtle commands with the appropriate functions and adding necessary indentation.